re/Field.re

```diff
@@ -0,0 +1,41 @@
+module type Config = {
+  type fields;
+  type state;
+  type schema;
+  let lens: list((fields, state => Value.t, (state, Value.t) => state));
+};
+
+module Make = (Config: Config) => {
+  type state =
+    | Validating
+    | Valid
+    | Error(string)
+    | Dirty
```

Comment:
I'm thinking that `Dirty` is redundant as there is a `Valid` (which means the field changed and was validated)

what you guys think? @Astrocoders/core

Reply:
yeah, but we have also the case where the user started to input but the field will be validated just after the submit, in this case the state dirty will be useful. Can be the state here the user started to type, but we don't want valid until he finish

Reply:
I agree with @medson10